### DISCUSSION TITLE
Getting error-bitmask 32, overheat

### DISCUSSION THREAD

**Discussion by odroegehorn** [syssi/esphome-soyosource-gtn-virtual-meter#147] [Q&A]
Hi all,
I'm getting with my 6 Soyos every now and then (at least 6 times an hour) the error-bitmask 32, telling overheat, without the fan running, and the temperatur of the sooy been reported below 40°C. 
Is it an information from the eps8266, or where does it come from?
It seems to reboot/restart and the, after around 10 seconds, it comes back to normal operation.

Any ideas?

I have already deactivated the captive portal and the web-server in the esp, just in case.

Thanks for help,
cheers.

**Comment by Chickenbreast0** [comment]
Hi,

It's interesting that you're writing this.
I had the problem 3 days ago that my inverter showed oscillating temperature spikes after a rebuilding of my setup (not the inverter itself). 

![image](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/assets/126453395/c4ebe2c6-2bed-4a0a-b69e-53a49a42e319)

A "SoftOff" of the inverter and a mains disconnection did not help.

I simply flashed the already existing *.YAML code on the ESP again and it worked as usual afterwards.
I cannot reproduce the reason for this. In my opinion, however, this is not due to the code itself. 
It may be due to your ESP8266. Try an ESP32, as it has more power. (Exclusion procedure)

**Reply by odroegehorn** [reply]
I don't have an ESP32, and especially not with an RS485 board attached to it.
With the Version 1.1.0 I never had this problem.

Maybe it's due to workload of the ESP. But maybe it's also due to lost data-packets from the soyo to the ESP and mis interpreted data?

The temp of the soyo itself is pretty constant for me.

**Reply by syssi** [reply]
How do you talk to the Soyo / which device model do you own?

**Reply by odroegehorn** [reply]
GTN-1200 with USB-Wifi Dongle, without Display.
I've soldered the RS-485 into the wifi-dongle, as described elsewhere here.
Works most of the time pretty well, except, when overheat happens.

**Reply by syssi** [reply]
You should avoid the OEM WiFi dongle if possible. The second microcontroller on the board (used as level shifter and energy meter) crashs quite often. On bootup this microcontroller sends some garbage to the ESP. The implemention tries to identify these reboots to discard bogus measurements. All these issues can be ruled if you avoid the dongle. I guess the temperature issue is gone too if you talk to the Soyosource directly using a plain ESP8266.

**Comment by syssi** [comment]
> Is it an information from the eps8266, or where does it come from?

The status frame of the Soyosource contains an error bitmask / error flags. One bit idendicates the `Overheat` error:

https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/components/soyosource_display/soyosource_display.cpp#L194

The bit is translated here into a string:

https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/components/soyosource_display/soyosource_display.cpp#L741C32-L755
https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/components/soyosource_display/soyosource_display.cpp#L29

These could be the root cause:

1. The Soyosource really reports the `Overheat` error for unknown reasons.
2. May be the interpretation of Bit 6 is wrong. If I remember correctly I've reverse engineered all values properly be using a emulated inverter.
3. The WiFi dongle doesn't tell the truth because of the unstable microcontroller.

The most likely cause is the first option unfortunately. Does the inverter stops to work if the `Overheat` error is shown? If yes, the inverter knows about the error.

**Comment by odroegehorn** [comment]
Well, 
what I see is the following:
The error get's reported, then the power_demand drops to zero, and THEN the power_output goes to zero.
My guess:
The Inverter has no idea, and the error comes from somewhere else.

Can you build a test-version, where the error 32 is ignored in the logic of the ESP?
As far as I understood, you only process the power_demand, when the error-bitmask is 0, right?

Maybe you could make a switch in the yaml file, to select to ignore error 32?

I know it's dangerous, but for testing purpose, this would be great.

I would love to stick to the wifi-dongle, as it fits quite nicely with the RS485 board.
Otherwise I need a level-shifter, and so on.

Thanks

**Reply by syssi** [reply]
So `soyosource_virtual_meter` component desn't care about the error bitmask. The component consumes the operation status only if it's passed into the component. Please remove this line to ignore the inverter feedback completely:

https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/esp8266-wifi-dongle-limiter-example.yaml#L83

**Comment by odroegehorn** [comment]
This seemed to work around the problem for now ....

**Reply by syssi** [reply]
Could you monitor the error bitmask sensor nevertheless? Is the `Overheat` / `value 32` issue gone or does the value doesn't affect the limiter anymore?

**Comment by odroegehorn** [comment]
Hi there,
so far, I've not seen the error/bitmask again. Seems to be gone.
It now only appears during boot-up of the soyo before it get's ready.

But beyond that, it's gone.
